ethers-providers/src/transports/quorum.rs

```diff
@@ -685,6 +682,95 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_get_quorum_block_number() {
+        let mut providers = Vec::new();
+        let mut mocked = Vec::new();
+
+        for value in [100, 101, 68, 100, 102] {
+            let mock = MockProvider::new();
+            for _ in 0..6 {
+                mock.push(U64::from(value)).unwrap();
+            }
+            providers.push(WeightedProvider::new(mock.clone()));
+            mocked.push(mock);
+        }
+
+        let quorum = QuorumProvider::builder()
+            .add_providers(providers.clone())
+            .quorum(Quorum::ProviderCount(5))
+            .build();
+        assert_eq!(quorum.get_quorum_block_number().await.unwrap().as_u64(), 68);
+
+        let quorum = QuorumProvider::builder()
+            .add_providers(providers.clone())
+            .quorum(Quorum::ProviderCount(4))
+            .build();
+        assert_eq!(quorum.get_quorum_block_number().await.unwrap().as_u64(), 100);
+
+        let quorum = QuorumProvider::builder()
+            .add_providers(providers.clone())
+            .quorum(Quorum::ProviderCount(3))
+            .build();
+        assert_eq!(quorum.get_quorum_block_number().await.unwrap().as_u64(), 100);
+
+        let quorum = QuorumProvider::builder()
+            .add_providers(providers.clone())
+            .quorum(Quorum::ProviderCount(2))
+            .build();
+        assert_eq!(quorum.get_quorum_block_number().await.unwrap().as_u64(), 101);
+
+        let quorum = QuorumProvider::builder()
+            .add_providers(providers.clone())
+            .quorum(Quorum::ProviderCount(1))
+            .build();
+        assert_eq!(quorum.get_quorum_block_number().await.unwrap().as_u64(), 102);
+
+        let quorum = QuorumProvider::builder()
+            .add_providers(providers.clone())
+            .quorum(Quorum::Majority)
+            .build();
+        assert_eq!(quorum.get_quorum_block_number().await.unwrap().as_u64(), 100);
+    }
+
+    #[tokio::test]
+    async fn test_get_quorum_block_number_with_errors() {
+        let mut providers = Vec::new();
+        let mut mocked = Vec::new();
+
+        let mock = MockProvider::new();
+        for _ in 0..2 {
+            mock.push(U64::from(100)).unwrap();
+        }
+        providers.push(WeightedProvider::new(mock.clone()));
+        mocked.push(mock);
+
+        let mock = MockProvider::new();
+        // this one will error
+        providers.push(WeightedProvider::new(mock.clone()));
+        mocked.push(mock);
+
+        let mock = MockProvider::new();
+
+        for _ in 0..2 {
+            mock.push(U64::from(101)).unwrap();
+        }
+        providers.push(WeightedProvider::new(mock.clone()));
+        mocked.push(mock);
+
+        let quorum = QuorumProvider::builder()
+            .add_providers(providers.clone())
+            .quorum(Quorum::ProviderCount(2))
+            .build();
+        assert_eq!(quorum.get_quorum_block_number().await.unwrap().as_u64(), 100);
+
```

Comment:
While we're at it I think we should add a test case where `quorum.get_quorum_block_number()` doesn't find a quorum

Reply:
Done